### ISSUE TITLE
Resetting random seed in data loader for every epoch

### ISSUE THREAD

**Issue by mszuyx** [pytorch/pytorch#76572]
### 🚀 The feature, motivation and pitch

Hi pytorch team,

I wrote a custom data loader in the following manner:

```
class data_loader(torch.utils.data.Dataset):
    def __init__(self, ):
     
    def __getitem__(self, index):

    def __len__(self):

```

The hope is to reset the random seed inside **__getitem__()** as the training progresses (e.g., a different seed for each epoch). As the training examples are randomly generated inside **__getitem__()**, if we don't reset the random seed, these samples will always be the same across different epochs. (I heard if we don't use **worker_init_fn**, the random process will be the same across different workers too.)

I also want to practice curriculum learning, which requires the data loader to have the knowledge of the current training process% (i.e., _current_epoch/total_epoch_). 

Currently I couldn't find a way to pass information (training% or preferred random seed) into the **__getitem__()** nor the data_loader class for that matter. It seems like everything is set in stone once I create the data_loader object.

Am I missing an existing solution or this is yet to be done? Is there some hack I can use to get around this?

Thanks!
Yu Chen
